### PR TITLE
Convert session storage from java serialization to jackson serializat…

### PR DESCRIPTION
**Pull request by wtritch**
…ion with redis.

This is a port of https://github.com/SmarterApp/RDW_Reporting/pull/483 to RDW_Admin.

### CODE REVIEW

File: common/src/main/java/org/opentestsystem/rdw/admin/common/security/PermissionScope.java [hunk 1]
```diff
@@ -18,6 +18,11 @@
     public static final PermissionScope STATEWIDE = new PermissionScope();
 
     /**
+     * Empty permission scope flyweight
+     */
+    public static final PermissionScope EMPTY = new PermissionScope(false);
```
Comment (wtritch) [inline common/src/main/java/org/opentestsystem/rdw/admin/common/security/PermissionScope.java:23]: I ported the explicit statewide PermissionScope implementation from RDW_Reporting in order to have a clean no-arg constructor for the Permission class.

File: webapp/src/main/java/org/opentestsystem/rdw/admin/security/User.java [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
     private String lastName;
     private Map<String, Permission> permissionsById;
-    private String username;
-    private String password;
-    private Collection<? extends GrantedAuthority> authorities;
-    private boolean accountNonExpired;
-    private boolean accountNonLocked;
-    private boolean credentialsNonExpired;
-    private boolean enabled;
+    private Map<Long, GroupGrant> groupsById;
```
Comment (wtritch) [inline webapp/src/main/java/org/opentestsystem/rdw/admin/security/User.java:22]: I ported the RDW_Reporting User changes over.  I don't believe RDW_Admin populates or uses the groupsById property in the hotfix, but I'm a fan of consistency.

**Review by malaffoon — APPROVED** [review]
rubber-stamp

File: webapp/src/main/java/org/opentestsystem/rdw/admin/security/session/SamlModule.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.security.providers.ExpiringUsernameAuthenticationToken;
+import org.springframework.security.saml.SAMLCredential;
+import org.springframework.security.saml.parser.SAMLCollection;
+import org.springframework.security.saml.parser.SAMLObject;
+
+/**
+ * This module registers de/serialization handlers for all classes
+ * serialized by SAML in the session store.
+ */
+public class SamlModule extends SimpleModule {
```
Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/admin/security/session/SamlModule.java:14]: some of this dup code going to find its way into common after the fix?

Comment (wtritch) [reply webapp/src/main/java/org/opentestsystem/rdw/admin/security/session/SamlModule.java:14]: I want to pull this into a web-base module once we move RDW_Admin into RDW_Reporting

**Pull request merged**